cloudrunv2: add support to startupProbe and livenessProbe to worker pool

victorsantos-cit opened the pull request:
<!--
Complete the self-review checklist to help speed up the review process: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/

If your PR is still work in progress, please create it in draft mode.

Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to.
For example: Fixes https://github.com/hashicorp/terraform-provider-google/issues/ISSUE_ID
-->
Hello Folks.
This PR is to fix https://github.com/hashicorp/terraform-provider-google/issues/23825

**Release Note Template for Downstream PRs (will be copied)**

See [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) for guidance.

```release-note:enhancement
cloudrunv2: added `startup_probe` and `liveness_probe` to `google_cloud_run_v2_worker_pool` resource
```

slevenick reviewed (changes requested):
Please reference the Terraform names of the fields in the changelog. They should be snake_case instead of camelCase. These changelogs are meant for users of the Terraform provider, not direct API users

victorsantos-cit commented:
Thanks for catching that! I've updated the changelog to use the Terraform snake_case field names instead of the API camelCase ones.

slevenick reviewed (changes requested):
Please address prior feedback

victorsantos-cit commented:
I change many, could you please validate again?

slevenick commented on mmv1/products/cloudrunv2/WorkerPool.yaml:
```diff
@@ -486,6 +486,140 @@ properties:
+                    Optional. TCPSocket specifies an action involving a TCP port. Exactly one of httpGet, tcpSocket, or grpc must be specified.
+                  type: NestedObject
+                  properties:
+                    - name: 'port'
```
Needs description

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/bc4d438392491a58ac37e812756412f1961f7707..f95111f0be72a17bcbd5342cce4ca598899c27ce) ( 4 files changed, 1642 insertions(+), 184 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/333b06a4d76bb167cbc4f0f23bdf50b8c68665ac..417d8a2ccdb74ee6734324ea58bccc506a1a6a14) ( 4 files changed, 1642 insertions(+), 184 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/21362e5a62e258825801e56929adee8dad17b870..65ea4e0be53c4aaeb8fb10d480cca4b1ae2ec48e) ( 1 file changed, 432 insertions(+))
Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/ce52376a929e5aaa9fccfad72140012947f2147f..9681a174eb305935c9308bf234169f7e2ca80059) ( 4 files changed, 163 insertions(+))



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_cloud_run_v2_worker_pool` (31 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_cloud_run_v2_worker_pool" "primary" {
  template {
    containers {
      liveness_probe {
        grpc {
          port    = # value needed
          service = # value needed
        }
        http_get {
          http_headers {
            port  = # value needed
            value = # value needed
          }
        }
        tcp_socket {
          port = # value needed
        }
      }
      startup_probe {
        grpc {
          port    = # value needed
          service = # value needed
        }
        http_get {
          http_headers {
            port  = # value needed
            value = # value needed
          }
          path = # value needed
          port = # value needed
        }
      }
    }
  }
}


```

slevenick reviewed (commented):
Please fix the test and run tests locally before requesting review

victorsantos-cit commented:
Sorry for all these errors, I adjusted everything you asked me, I added all the descriptions based on the API, I also adjusted the test and ran it locally and it worked, About the failure_threshold resource, if it doesn't pass, it sets its default, configured in the yaml, using the value set in the API as default.

modular-magician commented:
🟢 Tests passed during RECORDING mode:
`TestAccCloudRunV2WorkerPool_cloudrunv2WorkerPoolStartupLivenessProbeExample` [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15156/artifacts/ce4baeb0-4eee-43d1-80b5-427591725690/recording/TestAccCloudRunV2WorkerPool_cloudrunv2WorkerPoolStartupLivenessProbeExample.log)]

🟢 No issues found for passed tests after REPLAYING rerun.

---






🟢 All tests passed!

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-15156/artifacts/ce4baeb0-4eee-43d1-80b5-427591725690/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-15156/artifacts/ce4baeb0-4eee-43d1-80b5-427591725690/recording) for each test

kminsu-google commented:
@yanweiguo I think this is okay to go in.
